llvm/include/llvm/Analysis/ScalarEvolution.h

```diff
@@ -871,6 +871,13 @@ class ScalarEvolution {
   const SCEV *getExitCount(const Loop *L, const BasicBlock *ExitingBlock,
                            ExitCountKind Kind = Exact);
 
+  /// Same as above except this uses the predicated backedge taken info and
+  /// may require predicates.
+  const SCEV *
+  getPredicatedExitCount(const Loop *L, const BasicBlock *ExitingBlock,
+                         SmallVector<const SCEVPredicate *, 4> *Predicates,
```

Comment:
```suggestion
                         SmallVectorImpl<const SCEVPredicate *> *Predicates,
```

Reply:
Seems a reasonable change! While fixing this perhaps I should also fix up `getPredicatedBackedgeTakenCount` and `getPredicatedSymbolicMaxBackedgeTakenCount` too, because they currently take `SmallVector<const SCEVPredicate *, 4>` arguments.

Reply:
Done. I've also posted a NFC patch (https://github.com/llvm/llvm-project/pull/105663) to clean up the existing functions too.